lxd/storage/utils.go

```diff
@@ -970,6 +970,45 @@ func InstanceContentType(inst instance.Instance) drivers.ContentType {
 	return contentType
 }
 
+// volumeIsUsedByDevice; true when vol is referred to by dev
+// inst is the instance dev belongs to, or nil if dev is part of a profile.
+func volumeIsUsedByDevice(vol *api.StorageVolume, inst *db.InstanceArgs, dev map[string]string) (bool, error) {
```

Comment:
Shall we avoid passing vol as a pointer to avoid the implication/possibility for this function being able to modify it?